### ISSUE TITLE
Support libpurple 3 (unreleased)

### ISSUE THREAD

**Issue by je-vv** [majn/telegram-purple#366]
When building against libpurple/pidgin 3, 1st I needed to patch "configure" so that pkg-config asked for "purple-3" rather than purple, but that's the working part, :-).

Then, if building the master git version, compiling fails at:

+++
telegram-base.c: In function ‘tgp_error_if_false’:
telegram-base.c:240:5: warning: implicit declaration of function ‘purple_connection_error_reason’ [-Wimplicit-function-declaration]
     purple_connection_error_reason (conn->gc, PURPLE_CONNECTION_ERROR_AUTHENTICATION_FAILED, msg);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
telegram-base.c:241:5: error: too few arguments to function ‘purple_notify_message’
     purple_notify_message (_telegram_protocol, PURPLE_NOTIFY_MSG_ERROR, cause, msg, NULL, NULL, NULL);
+++

And if building on the development (1.4.0) branch, compiling fails at:

+++
In file included from telegram-purple.h:54:0,
                 from tgp-net.c:43:
tgp-chat.h:49:65: error: unknown type name ‘PurpleTypingState’
 unsigned int tgprpl_send_chat_typing (PurpleConversation *conv, PurpleTypingState typing, gpointer ignored);
                                                                 ^~~~~~~~~~~~~~~~~
tgp-net.c: In function ‘net_on_connected_assert_success’:
tgp-net.c:281:5: warning: implicit declaration of function ‘purple_connection_error_reason’ [-Wimplicit-function-declaration]
     purple_connection_error_reason (tls_get_conn (TLS), PURPLE_CONNECTION_ERROR_NETWORK_ERROR, _("Cannot connect to main server"));
+++

It seems like telegram-purple still doesn't support libpurple/pidgin 3.  Understood it's not yet released, but some (like me) need version 3.  If there's any branch able to support libpurple/pidgin 3, please let me know...

Thanks,

**Comment by je-vv** [comment]
Realized not that urgent...  Sorry...

**Comment by BenWiederhake** [comment]
True, not urgent.  But we will want purple-3 support *eventually*.

**Comment by BenWiederhake** [comment]
Good question.  Presumably updating the structures that interface tgp and libpurple.

If you do any modifications, please keep it intact for libpurple-2.  You're welcome to extend the tests to ensure this.
